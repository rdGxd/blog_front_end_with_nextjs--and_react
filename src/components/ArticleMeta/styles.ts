import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ", ";
    }

    .categories span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(50%);
    }
  `}
`;
