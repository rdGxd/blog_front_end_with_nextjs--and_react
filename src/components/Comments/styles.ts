import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
