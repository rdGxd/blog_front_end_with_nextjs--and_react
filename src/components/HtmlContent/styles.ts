import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(80%);
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      text-align: center;
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      line-height: 1.3;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium} 0;
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
