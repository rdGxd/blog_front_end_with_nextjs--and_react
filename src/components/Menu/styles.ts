import styled from "styled-components";

export const Wrapper = styled.div``;

export const Nav = styled.nav``;

export const Logo = styled.nav``;

export const OpenClose = styled.a``;
