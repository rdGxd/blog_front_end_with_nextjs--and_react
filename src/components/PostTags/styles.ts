import styled, { css } from "styled-components";

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    span {
      margin: 0 0rem 0 0.5rem;
    }

    span::before {
      content: " ";
    }

    span::after {
      content: ", ";
    }

    span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(80%);
    }
  `}
`;
