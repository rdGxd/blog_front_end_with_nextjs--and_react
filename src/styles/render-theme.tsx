import { BlogThemeProvider } from "../contexts/BlogThemeContext";
import { render, RenderResult } from "@testing-library/react";

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>);
};
