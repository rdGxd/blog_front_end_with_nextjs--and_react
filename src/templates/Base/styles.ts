import styled, { css } from "styled-components";

export const Wrapper = styled.div``;

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    ${theme.spacings.xhuge}
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xhuge};
  `}
`;

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
