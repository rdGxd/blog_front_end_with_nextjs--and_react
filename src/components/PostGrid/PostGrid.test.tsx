import { screen } from "@testing-library/react";
import { PostGrid } from ".";
import { renderTheme } from "../../styles/render-theme";

import mockPostGrid from "./mock";

describe("<PostGrid />", () => {
  it("should not render posts", () => {
    renderTheme(<PostGrid {...mockPostGrid} posts={undefined} />);

    expect(screen.getByText(/Nenhum Post/i)).toBeInTheDocument();
  });

  it("should render three posts", () => {
    const { container } = renderTheme(<PostGrid {...mockPostGrid} />);

    expect(screen.queryByText(/Nenhum post/i)).not.toBeInTheDocument();

    expect(screen.getAllByRole("heading")).toHaveLength(11);
    expect(screen.getAllByRole("img")).toHaveLength(11);
    expect(container.querySelectorAll("p")).toHaveLength(11);
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<PostGrid {...mockPostGrid} />);
    expect(container).toMatchSnapshot();
  });
});
