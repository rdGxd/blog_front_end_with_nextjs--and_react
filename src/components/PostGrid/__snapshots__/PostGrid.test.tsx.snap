// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PostGrid /> > should match snapshot 1`] = `
.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c2 .c4 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: 1.6rem;
}

.c2 a {
  color: inherit;
}

.c2:hover a {
  color: #dc143c;
}

.c2:hover img {
  opacity: 0.8;
}

.c3 {
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
}

.c0 {
  width: 100%;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

@media (max-width: 360px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          href="/post/6-papapa"
        >
          <img
            alt="6PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490199/360_F_571067620_JS_5_T5_Tk_Dtu3gf8_Wqm78_Ko_JRF_1vob_Pvo6_1dff1a52c6.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/6-papapa"
          >
            6PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/7-papapa"
        >
          <img
            alt="7PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490199/bf2191273cf0b9c83d0ee39819fe02b2_0b947ee9d4.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/7-papapa"
          >
            7PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/8-papapa"
        >
          <img
            alt="8PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490198/linhas_brilhantes_na_forma_3d_do_coracao_humano_em_fundo_escuro_generativo_ai_191095_1435_071d182d7c.webp"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/8-papapa"
          >
            8PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/9-papapa"
        >
          <img
            alt="9PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490199/ea8003eda835af024292148060887ed4_5f4ba76b6a.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/9-papapa"
          >
            9PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/2-papapa"
        >
          <img
            alt="2PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489865/1213_78bc74760f.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/2-papapa"
          >
            2PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum. Pellentesque convallis hendrerit est a auctor.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/10-papapa"
        >
          <img
            alt="10PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490198/baixados_1_a01f56f752.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/10-papapa"
          >
            10PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/1-papapa"
        >
          <img
            alt="1PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489672/baixados_58294ddddf.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/1-papapa"
          >
            1PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/3-papapa"
        >
          <img
            alt="3PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489959/1325363_6dfc0b7add.png"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/3-papapa"
          >
            3PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus porta augue quis ligula vehicula elementum. Pellentesque convallis hendrerit est a auctor.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/4-papapa"
        >
          <img
            alt="4PAPAPA"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490202/uma_pintura_de_um_lago_de_montanha_com_uma_montanha_ao_fundo_188544_9126_bc5fa577b5.webp"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/4-papapa"
          >
            4PAPAPA
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
      <div
        class="c2"
      >
        <a
          href="/post/5-papa-pa"
        >
          <img
            alt="5PAPAPa"
            class="c3"
            src="https://res.cloudinary.com/dccldzodg/image/upload/v1700490200/pexels_pixabay_36717_154aaae2b0.jpg"
          />
        </a>
        <h2
          class="c4 c5"
        >
          <a
            href="/post/5-papa-pa"
          >
            5PAPAPa
          </a>
        </h2>
        <p
          class=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
      </div>
    </div>
  </div>
</div>
`;
